modules/editor-service/src/main/java/org/opencastproject/editor/EditorServiceImpl.java

```
@@ -437,6 +447,82 @@ void addSmilToArchive(MediaPackage mediaPackage, final Smil smil) throws IOExcep
 
   }
 
+  /**
+   * Adds subtitles {@link PostEditingData.Subtitle} to the media package and sends the updated media package
+   * to the archive. If a subtitle flavor already exists, the subtitle is overwritten
+   *
+   * @param mediaPackage
+   *          the media package to at the SMIL catalog
+   * @param subtitles
+   *          the subtitles to be added
+   * @throws IOException
+   */
+  private void addSubtitleTrack(MediaPackage mediaPackage, List<PostEditingData.Subtitle> subtitles)
+          throws IOException {
+    for (PostEditingData.Subtitle subtitle : subtitles) {
+      if (!subtitle.getFlavor().matches(captionsFlavor)) {
+        break;
+      }
```

Maybe raise an exception which ends in a 400 Bad Request for the user, so it is clear that the request wasn't processed successfully and, in particular, wasn't saved as sent by the user. Also, maybe not save anything if half of the request is bad? Otherwise, you don't know the state you end up with.